lib/process-services-cloud/src/lib/form/components/attach-file-cloud-widget/attach-file-cloud-widget.component.ts

```diff
@@ -0,0 +1,151 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* tslint:disable:component-selector */
+
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import {
+  FormService,
+  LogService,
+  ThumbnailService,
+  ProcessContentService
+} from '@alfresco/adf-core';
+import { RelatedContentRepresentation } from '@alfresco/js-api';
+import { ContentCloudNodeSelectorService } from '../../services/content-cloud-node-selector.service';
+import { ProcessCloudContentService } from '../../services/process-cloud-content.service';
+import { UploadCloudWidgetComponent } from '../upload-cloud.widget';
+
+@Component({
+  selector: 'adf-cloud-attach-file-cloud-widget',
+  templateUrl: './attach-file-cloud-widget.component.html',
+  styleUrls: ['./attach-file-cloud-widget.component.scss'],
+  host: {
+    '(click)': 'event($event)',
+    '(blur)': 'event($event)',
+    '(change)': 'event($event)',
+    '(focus)': 'event($event)',
+    '(focusin)': 'event($event)',
+    '(focusout)': 'event($event)',
+    '(input)': 'event($event)',
+    '(invalid)': 'event($event)',
+    '(select)': 'event($event)'
+  },
+  encapsulation: ViewEncapsulation.None
+})
+export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent implements OnInit {
+
+  static ACS_SERVICE = 'alfresco-content';
+  tempFilesList = [];
+
+  constructor(
+    public formService: FormService,
+    public logger: LogService,
+    public processContentService: ProcessContentService,
+    public thumbnails: ThumbnailService,
+    public processCloudContentService: ProcessCloudContentService,
+    public contentNodeSelectorService: ContentCloudNodeSelectorService) {
+    super(formService, thumbnails, processCloudContentService, logger);
+  }
+
+  ngOnInit() {
+    if (this.field &&
+      this.field.value &&
+      this.field.value.length > 0) {
+      this.tempFilesList = [...this.field.value];
+      this.hasFile = true;
+    }
+    this.getMultipleFileParam();
+  }
+
+  isFileSourceConfigured(): boolean {
+    return !!this.field.params && !!this.field.params.fileSource;
+  }
+
+  isMultipleSourceUpload(): boolean {
+    return !this.field.readOnly && this.isFileSourceConfigured() && !this.isOnlyLocalSourceSelected();
+  }
+
+  isOnlyLocalSourceSelected(): boolean {
+    return this.field.params &&
+      this.field.params.fileSource &&
+      this.field.params.fileSource.serviceId === 'local-file';
+  }
+
+  isSimpleUploadButton(): boolean {
+    return this.isUploadButtonVisible() &&
+      !this.isFileSourceConfigured() ||
+      this.isOnlyLocalSourceSelected();
+  }
+
+  isUploadButtonVisible(): boolean {
+    return (!this.hasFile || this.multipleOption) && !this.field.readOnly;
+  }
+
+  onAttachFileChanged(event: any) {
+    this.tempFilesList.push(...Array.from(event.target.files));
```

Comment:
i don't think you still need this tempFile array. This was necessary for showing the preview and download of a just uploaded node.